src/AppInstallerSharedLib/Public/AppInstallerVersions.h

```diff
@@ -119,12 +119,25 @@ namespace AppInstaller::Utility
 
         std::string m_version;
         std::vector<Part> m_parts;
+        bool m_trimPrefix = true;
         ApproximateComparator m_approximateComparator = ApproximateComparator::None;
 
       // Remove trailing empty parts (0 or empty)
         void Trim();
     };
 
+    // Version that does not have leading non-digit characters trimmed
+    struct RawVersion : protected Version
+    {
+        RawVersion() { m_trimPrefix = false; }
+
+        RawVersion(const std::string& version, std::string_view splitChars = DefaultSplitChars) :
+            RawVersion(std::string(version), splitChars) { m_trimPrefix = false; }
+        RawVersion(std::string&& version, std::string_view splitChars = DefaultSplitChars);

```

Comment:
nit: just taking `std::string version` if you needed the memory for yourself is the simplest way.  l-values will copy construct (as you are doing manually) and r-values will move construct.

nit2: the `string&` constructor doesn't need to set `m_trimPrefix` again.

Reply:
I see. . .  I was just copying what `Version` had done, so wasn't paying attention to how the values actually get handled in memory. I presume the easiest is just to remove the `string&` constructor?

Reply:
Remove the `string&` and change `string&&` to `string`.